### ISSUE TITLE
Music Player stops when used with the Headphone Detector App

### ISSUE THREAD

**Issue by collectorgeneral** [vanilla-music/vanilla#269]
As the title says, quite a few times when plugging the headset, the music player would launch, play the music for a second, stutter and stop playing altogether. When the screen is unlocked, it asks to launch the search app for some reason. When its locked, the music would just stutter and stop playing after like 1 second. Attached a video in case you(dev) want to see it. 
[Telecine_2016-01-18-10-25-39.zip](https://github.com/vanilla-music/vanilla/files/94520/Telecine_2016-01-18-10-25-39.zip)

Additional information:
Device: One Plus One (aka Bacon aka A0001), Running Cyanogen OS 12.1.1 - YOG7DAS2K1, Android 5.1.1

**Comment by adrian-bl** [comment]
Is this a new behavior or has it done this since ever?

> When the screen is unlocked, it asks to launch the search app for some reason

There is no way we could trigger this: The headphone detection code is pretty simple:
https://github.com/vanilla-music/vanilla-headphone-detector/blob/master/src/ch/blinkenlights/android/vanillaplug/VPlugService.java#L91

All it does is sending an ch.blinkenlights.android.vanilla.action.PLAY event which will be captured by vanilla music (causing it to start)

Are you sure that no other app attempts to capture the headset-insert event?
Did you verify that cyanogenmods own headset-detection (Settings -> Sound -> Launch music app) is off?

**Comment by collectorgeneral** [comment]
Yes I have checked the setting. They are what you suggested they should be. But despite that it was displaying that odd behaviour.
I went ahead and restarted my phone. Everything went back to normal. However, I suspect its either VLC or Timber Music player. I tried launching VLC and it made my phone hang up and started playing music despite the fact that I did not ask it to and it displayed the exact same behaviour. I had to try and hard restart again and now it works again. I had similar issue with Timber Music player a while ago where I accidentally long pressed my volume rocker and it launched vanilla and they both were running at the same time and caused my phone to hang up. Once I restarted my phone everything went back to normal and the issue didnt crop up again. Not sure whats causing all the mess. I will let you know if I am able to reproduce the issue again.
